wordgenerator.py

```diff
@@ -0,0 +1,110 @@
+import pygame
+import random
+import sys
+import math
+from copy import deepcopy
+
+CONST_FILENAME = "american-english.txt"  
+
+'''
+Opens the file CONST_FILENAME and reads entire file to word_array list. Shuffles the array and then sorts according to length
+'''
+class WordGenerator():
+    index_locations = []
+    index_prime_locations = []
+    word_array = []
+    number_of_lengths = 0
+    smallest_length = []
+
+    def __init__(self):
+        with open(CONST_FILENAME) as infile:
+            for line in infile:
+                self.word_array.append(line.rstrip('\n'))
+        random.shuffle(self.word_array) 
+        self.word_array.sort(key=len)
+        self.split_list(self.word_array)
+
+    '''
+    Will populate the list index_locations on where each new length starts
+    For example, index_locations[3] will give you the location of the first word with length 3
+    index_locations[4] will give you location of first word with length 4, etc.
+    index_prime_locations will be incremented and used to keep track of which words have been given out.
+    '''
+    def split_list(self, list):
+        self.number_of_lengths = len(list[len(list)-1]) - len(list[0]) 
```

Comment:
`len(list[len(list)-1]) - len(list[0])`  
This line is slightly cryptic. 

You also never use `self.number_of_lengths` outside of this function so I don't think it should be a member variable.

Since you calculate `self.smallest_length` anyways I'd suggest replacing these two lines with.

```
shortest_word_length = len(list[0])
longest_word_length = len(list[:-1]) # [:-1] is the pythonic way of getting the last item in a list
number_of_lengths = longest_word_length = shortest_word_length 
```
I think `shortest_word_length` is a better (but only slightly better) name than smallest_length.